### ISSUE TITLE
Mixing types does not work

### ISSUE THREAD

**Issue by aplavin** [queryverse/Query.jl#279]
Hi! There is a very confusing error thrown in some cases with mixed types.

A minimal example:
```julia
    df = DataFrame(a=1:1)
    df |>
        @mapmany([(c=Inf,), (c=1,)], (; _..., b=__)) |>
        @groupby({_.b}) |>
        @map((; key(_)..., a=mean(_.a)))
```
throws `type NamedTuple has no field a` on the last line with `@map`. This is due to `c` having type `Float` in one row, and `Int` in another. When all values have the same type, the same code works:
```julia
    df = DataFrame(a=1:1)
    df |>
        @mapmany([(c=0,), (c=1,)], (; _..., b=__)) |>  # changed Inf to 0
        @groupby({_.b}) |>
        @map((; key(_)..., a=mean(_.a)))
```

Julia 1.2.0, Query v0.12.1.

**Comment by davidanthoff** [comment]
Yeah, the current design _very much_ only works for "type stable iterators", i.e. iterators where each element has the same type. I'll have to think whether we can relax that.

For now a better error message would of course be helpful, but that might also not be easy...

**Comment by aplavin** [comment]
Oh, didn't know that, and didn't see anywhere in the docs. Well, this is a pretty strong limitation to keep in mind...
